### PR TITLE
Configuring --gcp-project-type and --runtime-config

### PR DESCRIPTION
**Pull request by Namanl2001**
Configuring `--gcp-project-type` and `--runtime-config`

cc: @amwat @dims

### CODE REVIEW

**Comment by Namanl2001** [comment]
Corresponding PR in K/K https://github.com/kubernetes/kubernetes/pull/106431

File: pkg/testers/node/node.go [hunk 1]
```diff
@@ -61,6 +60,8 @@ type Tester struct {
 	ImageConfigFile                string `desc:"Path to a file containing image configuration."`
 	ImageConfigDir                 string `desc:"Path to image config files."`
 	Parallelism                    int    `desc:"The number of nodes to run in parallel."`
+	GcpProjectType                 string `desc:"Explicitly indicate which project type to select from boskos."`
```
Comment (amwat) [inline pkg/testers/node/node.go:63]: nit: use `GCPProjectType` instead of `GcpProjectType`

File: pkg/testers/node/node.go [hunk 2]
```diff
@@ -242,6 +244,7 @@ func (t *Tester) constructArgs() []string {
 		"IMAGE_CONFIG_DIR=" + t.ImageConfigDir,
 		"SSH_USER=" + t.sshUser,
 		"SSH_KEY=" + t.privateKey,
+		"RUNTIME_CONFIG=" + t.RuntimeConfig,
```
Comment (amwat) [inline pkg/testers/node/node.go:247]: does it work correctly if it empty?

**Review by amwat — COMMENTED** [review]
/lgtm
/approve
/hold

File: pkg/testers/node/node.go (hunk 2 above)
Comment (Namanl2001) [reply pkg/testers/node/node.go:247]: Thanks for catching it out. This needs to be checked explicitly as it's been done for kubetest https://github.com/kubernetes/test-infra/blob/8c944aa6aa17c1efd224629b250e9bd52eeb7132/kubetest/e2e.go#L625-L627

File: pkg/testers/node/node.go (hunk 1 above)
Comment (Namanl2001) [reply pkg/testers/node/node.go:63]: 👍 output would be same for both, but `GCPProjectType` seems nicer.

**Pull request merged**